src/utils/generator.js

```diff
@@ -1123,6 +1179,22 @@ export class Generator {
       dependencies: [generator1, generator2],
     });
   }
+
+  replaceEntry(entry: GeneratorEntry, replacement: GeneratorEntry) {
```

Comment:
Replacing entries is now problematic, as you now keep around a map from args to entries.

So, let's scrap this whole generator entry replacement idea and instead go back to in-place mutation for the moment. Replacing whole generator entries wasn't really what I was after to begin with...

What happens here is that the "dependencies" are mutated (and at some point the visitor should take into account such dependencies when iterating for fixpoint). What I'll really need in the future is that all such dependency mutations get tracked somehow. But I am fine to just ignore that for, as long as the mutation is easily discoverable in the `visit` method.